halo2_proofs/src/multicore.rs

```diff
@@ -1,5 +1,74 @@
-//! An interface for dealing with the kinds of parallel computations involved in
-//! `halo2`. It's currently just a (very!) thin wrapper around [`rayon`] but may
-//! be extended in the future to allow for various parallelism strategies.
+#[cfg(all(
+    feature = "multicore",
+    target_arch = "wasm32",
+    not(target_feature = "atomics")
+))]
+compile_error!(
+    "The multicore feature flag is not supported on wasm32 architectures without atomics"
+);
 
-pub use rayon::{current_num_threads, scope, Scope};
+pub use maybe_rayon::{
+    iter::{
+        IndexedParallelIterator, IntoParallelIterator, IntoParallelRefIterator,
+        IntoParallelRefMutIterator, ParallelIterator,
```

Comment:
Just realized some issue here, will submit a fix soon

Reply:
Fixed in https://github.com/privacy-scaling-explorations/halo2/pull/199/commits/22605f5b1e8d94a4a0b2959a490976ebed1b534c.